## Manoj Mahanty - Jr. Engineer - Data Analytics
####  Python | MATLAB | R Begineer | SQL, PostgreSQL | MS Excel

![](https://media-exp1.licdn.com/dms/image/C4E03AQHQwbwhrso79w/profile-displayphoto-shrink_100_100/0/1630124536553?e=1635379200&v=beta&t=VSAf_0U1A_gCd9m1GEN_UMYQTiCgumibjqYsfXp2_VM)  
 
* email: mahantymanoj786@gmail.com
* LinkedIn : https://www.linkedin.com/in/manoj-mahanty-aa3769115/

### Python was conceived in the late 1980s[38] by Guido van Rossum at Centrum Wiskunde & Informatica (CWI) in the Netherlands as a successor to ABC programming language, which was inspired by SETL,[39] capable of exception handling and interfacing with the Amoeba operating system.[10] Its implementation began in December 1989.[40] Van Rossum shouldered sole responsibility for the project, as the lead developer, until 12 July 2018, when he announced his "permanent vacation" from his responsibilities as Python's
source https://en.wikipedia.org/wiki/Python_(programming_language)

![](https://i.imgur.com/gvSnw4A.png)

This tutorial covers the following topics:

* Performing arithmetic operations using Python
* Solving multi-step problems using variables
* Evaluating conditions using Python
* Combining conditions with logical operators
* Adding text styles using Markdown

Running the Jupyter Notebook
press Shift + Enter to execute the cell

Run the code cells below to perform calculations and view their result. Try changing the numbers and run the modified cells again to see updated results. Can you guess what the `//`, `%`, and `**` operators are used for?

## Arithmetic Operator 

![](https://i.imgur.com/PVOe4o9.jpeg)

In [1]:
2 + 3 + 9

14

In [2]:
99 - 73

26

In [3]:
23.54 * -1432

-33709.28

In [4]:
100 / 7

14.285714285714286

In [5]:
100 // 7

14

In [6]:
100 % 7

2

In [7]:
5 ** 3

125

As you might expect, operators like `/` and `*` take precedence over other operators like `+` and `-` as per mathematical conventions. You can use parentheses, i.e. `(` and `)`, to specify the order in which operations are performed.

In [8]:
((2 + 5) * (17 - 3)) / (4 ** 3)

1.53125

Python supports the following arithmetic operators:

| Operator   | Purpose           | Example     | Result    |
|------------|-------------------|-------------|-----------|
| `+`        | Addition          | `2 + 3`     | `5`       |
| `-`        | Subtraction       | `3 - 2`     | `1`       |
| `*`        | Multiplication    | `8 * 12`    | `96`      |
| `/`        | Division          | `100 / 7`   | `14.28..` |
| `//`       | Floor Division    | `100 // 7`  | `14`      |    
| `%`        | Modulus/Remainder | `100 % 7`   | `2`       |
| `**`       | Exponent          | `5 ** 3`    | `125`     |

# Conditional Operator

## Evaluating conditions using Python

Apart from arithmetic operations, Python also provides several operations for comparing numbers & variables.

| Operator    | Description                                                     |
|-------------|-----------------------------------------------------------------|
| `==`        | Check if operands are equal                                     |
| `!=`        | Check if operands are not equal                                 |
| `>`         | Check if left operand is greater than right operand             |
| `<`         | Check if left operand is less than right operand                |
| `>=`        | Check if left operand is greater than or equal to right operand |
| `<=`        | Check if left operand is less than or equal to right operand    |

The result of a comparison operation is either `True` or `False` (note the uppercase `T` and `F`). These are special keywords in Python. Let's try out some experiments with comparison operators.

In [11]:
my_favorite_number = 1
my_least_favorite_number = 5
a_neutral_number = 3

In [12]:
# Equality check - True
my_favorite_number == 1

True

In [13]:
# Equality check - False
my_favorite_number == my_least_favorite_number

False

In [14]:
# Not equal check - True
my_favorite_number != a_neutral_number

True

In [15]:
# Not equal check - False
a_neutral_number != 3

False

In [16]:
# Less than check - True
my_favorite_number < 10

True

In [17]:
# Less than check - False
my_least_favorite_number < my_favorite_number

False

In [18]:
# Greater than or equal check - True
my_favorite_number >= 1

True

In [19]:
# Greater than or equal check - False
my_favorite_number >= 3

False

In [20]:
# Less than or equal check - True
3 + 6 <= 9

True

In [21]:
# Less than or equal check - False
my_favorite_number + a_neutral_number <= 3

False

## Combining conditions with logical operators

The logical operators `and`, `or` and `not` operate upon conditions and `True` & `False` values (also known as *booleans*). `and` and `or` operate on two conditions, whereas `not` operates on a single condition.

The `and` operator returns `True` when both the conditions evaluate to `True`. Otherwise, it returns `False`.

| `a`     | `b`    | `a and b` |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `False`   |
|  `False`| `True` | `False`   |
|  `False`| `False`| `False`   |


The `or` operator returns `True` if at least one of the conditions evaluates to `True`. It returns `False` only if both conditions are `False`.

| `a`     | `b`    | `a or b`  |
|---------|--------|-----------|
|  `True` | `True` | `True`    |
|  `True` | `False`| `True`    |
|  `False`| `True` | `True`    |
|  `False`| `False`| `False`   |




In [28]:
my_favorite_number

1

In [23]:
my_favorite_number > 0 and my_favorite_number <= 3

True

In [24]:
my_favorite_number < 0 and my_favorite_number <= 3

False

In [25]:
my_favorite_number > 0 and my_favorite_number >= 3

False

In [29]:
True and False

False

In [30]:
True and True

True

In [31]:
a_neutral_number == 3 or my_favorite_number < 0

True

In [32]:
a_neutral_number != 3 or my_favorite_number < 0

False

In [33]:
False or False

False

The `not` operator returns `False` if a condition is `True` and `True` if the condition is `False`.

In [34]:
not a_neutral_number == 3

False

In [35]:
not my_favorite_number < 0

True

In [36]:
not False

True

In [37]:
not True

False

Logical operators can be combined to form complex conditions. Use round brackets or parentheses `(` and `)` to indicate the order in which logical operators should be applied.

In [38]:
(2 > 3 and 4 <= 5) or not (my_favorite_number < 0 and True)

True

In [39]:
not (True and 0 < 1) or (False and True)

False

If parentheses are not used, logical operators are applied from left to right.

In [40]:
not True and 0 < 1 or False and True

False

# Questions for Revision

## 1) What is a Jupyter notebook?

The Jupyter Notebook App is a server-client application that allows editing and running notebook documents via a web browser. The Jupyter Notebook App can be executed on a local desktop requiring no internet access




## 2) What the different arithmetic operations supported in Python?

There are 7 arithmetic operators in Python :

Addition: + \
Subtraction: - \
Multiplication: * \
Division: / \
Modulus: % \
Exponentiation: ** \
Floor division: // \


## 3) What is the difference between the / and the // operators?

'/' - division eg: 4/3 = 1.3333333333333333 \
'//' - Floor Division eg : 4//3 = 1

4) What are the different ways of creating comments in Python code? \
5) What are the different comparison operations supported in Python? \
6) What is the result of a comparison operation? \
7) What is the difference between `=` and `==` in Python? \
8) What are the logical operators supported in Python? \
9) What is the difference between the `and` and `or` operators? \
10) Can you use comparison and logical operators in the same expression? \
11) What is the purpose of using parentheses in arithmetic or logical expressions? \
12) What is Markdown? Why is it useful? 